cloud-controller-manager/do/cloud.go

```diff
@@ -155,18 +155,25 @@ func (c *cloud) Initialize(clientBuilder cloudprovider.ControllerClientBuilder,
 	go res.Run(stop)
 	go c.serveDebug(stop)
 
+	// Tags are returned as a slice of godo.Tag but we want a slice of strings.
+	var tags []string
+	tagList, _, err := c.resources.gclient.Tags.List(ctx, opts)
+	if err != nil {
+		klog.Errorf("Failed to get tags: %s", err)
+	}
+	for _, t := range tagList {
+		tags = append(tags, t.Resources.Droplets.LastTaggedURI)
```

Comment:
Looking at this further. This isn't what I want. This is just the URI for the resource (in this case droplet) that was last tagged.

`Tags.List()` gives me a list of `godo.Tag`. The FirewallRulesRequest has a tags field as `[]string`. I'm going to look at what that `[]string` actually is and update this again. @timoreimann

Reply:
There is a `name` field on tag.

```go
// Tag represent DigitalOcean tag
type Tag struct {
	Name      string           `json:"name,omitempty"`
	Resources *TaggedResources `json:"resources,omitempty"`
}
```

Do we want a slice of tag name strings as "tags" passed into the NewFirewallController() and essentially FirewallRulesRequest 🤔

Reply:
**Update**: confirmed with the firewall team that `t.Name` is what we want